pkg/controller/vitesskeyspace/reconcile_shards.go

```diff
@@ -151,6 +152,27 @@ func updateVitessShard(key client.ObjectKey, vts *planetscalev2.VitessShard, vtk
 	// Update labels, but ignore existing ones we don't set.
 	update.Labels(&vts.Labels, newShard.Labels)
 
+	// Remove old annotations that shouldn't be there that we injected previously.
+	differentAnnotations := differentKeys(newShard.Annotations, vts.Spec.Annotations)
+	for _, annotation := range differentAnnotations {
+		delete(vts.Annotations, annotation)
+	}
+
+	// Update annotations we set.
+	update.Annotations(&vts.Annotations, newShard.Annotations)
+
 	// For now, everything in Spec is safe to update.
 	vts.Spec = newShard.Spec
 }
+
+// differentKeys returns keys from an older map instance that are no longer in a newer map instance.
+func differentKeys(oldMap, newMap map[string]string) []string {
+	var differentKeys []string
+	for k, _ := range oldMap {
```

Comment:
You can just leave out the value if you don't need it:

```suggestion
	for k := range oldMap {
```

Reply:
Done https://github.com/planetscale/vitess-operator/pull/51/commits/cfce0b22ddf5451c2b02a79f6083963644a2a65a.